.github/workflows/build-and-push-ghcr.yaml

```diff
@@ -0,0 +1,77 @@
+name: Build and Deploy
+
+on:
+  push:
+    branches:
+      - main
+      - test
+      - dev
+
+env:
+  DEPLOYMENT_NAME: visitz-api-deployment
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+  GH_ENVIRONMENT: ${{ github.ref_name }}
```

Comment:
@kiiskila-bcgov I think this is how we'll automatically use an environment per branch.